src/shared/contexts/pinneditems.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {
+  createContext,
+  FC,
+  useEffect,
+  useState,
+  memo,
+  useCallback,
+  ReactElement,
+} from 'react'
+
+export interface PinnedItem {
+  orgID: string
+  userID: string
+  id: string
+  createdAt: string
+  updatedAt: string
+  metadata: {}
+  type: string
+}
+
+export enum PinnedItemTypes {
+  Dashboard = 'dashboard',
+  Cell = 'cell',
+  Notebook = 'notebook',
+  Task = 'task',
+  Alert = 'alert',
+  UploadCSV = 'uploadCsv',
+  UploadLP = 'uploadLP',
+  ClientLibrary = 'clientLibrary',
+}
+export const PinnedItemsContext = createContext(null)
+
+const token = ''
+
+// The routes below are placeholders until we get the implementation of the swagger-defined routes through the pipeline
+
+const getPinnedItems = async () => {
+  return await fetch(``, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      authorization: `Bearer ${token}`,
+    },
+  })
+}
+
+const removePinnedItem = async (id: string) => {
+  return await fetch(`/${id}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+      authorization: `Bearer ${token}`,
+    },
+  })
+}
+const addPinnedItem = async (item: Partial<PinnedItem>) => {
+  const added = await fetch(``, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      authorization: `Bearer ${token}`,
+    },
+    body: JSON.stringify(item),
+  })
+  return await added.json()
+}
+
+const updatePinnedItem = async (_id: string, item: Partial<PinnedItem>) => {
+  await fetch(``, {
+    method: 'PUT',
+    headers: {
+      'Content-Type': 'application/json',
```

Comment:
I get that these are all placeholders, can we get a TODO with an issue linked to move this over to a generated route so we don't end up with UI code for this?

Reply:
yup, sure thing